### ISSUE TITLE
beagleboard: nand/omap2 module is not created if onenand/omap2 is

### ISSUE THREAD

**Issue by npf** [RobertCNelson/linux-dev#40]
Hello,
Using kernels from the rcn-ee Debian Jessie repository: 4.1.5-armv7-x2 or 4.2.0-rc6-armv7-x0, my beagleboard does not see its nand (mtd).
Those kernels have the onenand/omap2.ko driver but not the nand/omap2.ko.

Compiling a kernel without onenand (CONFIG_MTD_ONENAND not set) builds nand/omap2.ko, and the nand shows up on the system when loading the module.

Looks like having the same name for both modules causes the issue?

**Comment by RobertCNelson** [comment]
Yeah, that's a nasty bug. I have an am37xx based device with onenand..

I wonder if this is fixed with Roger's patchset here, if not can you ping him on this issue, so we can get it fixed upstream?

https://www.mail-archive.com/linux-kernel@vger.kernel.org/msg950586.html

Regards,
